store.go

```diff
@@ -440,6 +444,10 @@ type Store interface {
 	// associated with a layer.
 	LayerBigData(id, key string) (io.ReadCloser, error)
 
+	// LayerBigDataFile retrieves the backing file for the layers big data,
+	// when used.
+	LayerBigDataFilePath(id, key string) (string, error)
```

Comment:
This is the public API, so this is the place to document that it can return `""`.

But this is fundamentally racy. After this returns, the layer’s BigData can be added/modified/removed at any time, hypothetically changing what the path points to. Fine, we can document that this returns a non-empty string only if the path will only refer to an item for (`id, key`) and not for any other item.

But the whole layer can be removed after this returns. And recreated again. And the BigData item can be created, with different contents, afterwards. All of this could happen before “we” get to consume the returned path.

Also, some other process can _change_ the BigData item at any time, as long as “we” are not inside a `Store` method and holding a lock.

_Maybe_ one option would be to return a file descriptor, _and_ to guarantee that we use `Rename()` on BigData creation so that the file descriptor refers to immutable data.